src/modules/graphics/Texture.h

```diff
@@ -139,6 +139,10 @@ struct SamplerState
 	static bool getConstant(const char *in, WrapMode &out);
 	static bool getConstant(WrapMode in, const char *&out);
 	static std::vector<std::string> getConstants(WrapMode);
+
+	bool operator==(const SamplerState& other) const {
+		return memcmp(this, &other, sizeof(SamplerState)) == 0;
```

Comment:
I wonder if this will work correctly if the compiler adds any padding around the struct members

Reply:
VkSampler objects get created and cached by the Graphics class, and stored in a std::unordered_map, that's why this is needed. Although I think I will change it, such that the VkSampler gets recreated everytime the SamplerState changes, since that should make the code simpler. It shouldn't also be a performance problem since changing the SamplerState shouldn't happen every frame (usually only in initialization).

Reply:
I think the current cache approach is fine, I just suspect the `memcmp` instead of comparing individual fields might break because there might be uninitialized memory next to each struct field, if the compiler adds padding.

Reply:
Oh I just see there is a toKey method that's used by the metal backend that I have overlooked, I'll switch to that. Then the operator== method is not needed anymore.